mesonbuild/cmake/tracetargets.py

```diff
@@ -24,7 +24,7 @@ def resolve_cmake_trace_targets(target_name: str,
                                 trace: 'CMakeTraceParser',
                                 env: 'Environment',
                                 *,
-                                clib_compiler: T.Optional['Compiler'] = None,
+                                clib_compiler: T.Union['MissongCompiler', 'Compiler'] = None,
```

Comment:
> It should be imported in T.TYPE_CHECKING.
>
> But if it's supposed to be a MissingCompiler in this case, then it shouldn't default to None...

agreed.

> I think since this already handles None, you don't need to modify this file though.

We followed your first suggestion, because the point of the `MissingCompiler` object is to replace `None`